pkg/ui/serve.go

```diff
@@ -3,27 +3,67 @@ package ui
 //go:generate go run ../controller/kieapp/defaults/.packr/packr.go
 
 import (
-	"log"
-
 	"github.com/RHsyseng/console-cr-form/pkg/web"
 	"github.com/ghodss/yaml"
 	"github.com/go-openapi/spec"
 	"github.com/gobuffalo/packr/v2"
-	"github.com/sirupsen/logrus"
+	v1 "github.com/kiegroup/kie-cloud-operator/pkg/apis/app/v1"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/logs"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
 )
 
+var log = logs.GetLogger("ui")
+
 // Listen ...
 func Listen() {
-	config := &web.ConfigurationHolder{
-		PortField:   8080,
-		SchemaField: getSchema(),
-		FormField:   getForm(),
+	config, err := web.NewConfiguration("", 8080, getSchema(), v1.SchemeGroupVersion.String(), getObjectKind(), getForm(), apply)
+	if err != nil {
+		log.Fatal("Failed to configure web server", err)
 	}
 	if err := web.RunWebServer(config); err != nil {
-		logrus.Errorf("Failed to run web server: %v", err)
+		log.Fatal("Failed to run web server", err)
 	}
 }
 
+func apply(cr string) {
+	log.Debugf("Will deploy KieApp based on yaml %v", cr)
+	kieApp := &v1.KieApp{}
+	err := yaml.Unmarshal([]byte(cr), kieApp)
+	if err != nil {
+		log.Errorf("Failed to parse CR based on %s, error is %v", cr, err)
+		return
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Error("Failed to get in-cluster config", err)
+		return
+	}
+	err = v1.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Error("Failed to add scheme", err)
+		return
+	}
+	config.ContentConfig.GroupVersion = &v1.SchemeGroupVersion
+	config.APIPath = "/apis"
+	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+	config.UserAgent = rest.DefaultKubernetesUserAgent()
+	restClient, err := rest.UnversionedRESTClientFor(config)
+	if err != nil {
+		log.Error("Failed to get REST client", err)
+		return
+	}
+	kieApp.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("KieApp"))
+	err = restClient.Post().Namespace("operator-ui").Body(kieApp).Resource("kieapps").Do().Into(kieApp)
```

Comment:
Please provide correct namespace selection instead of "operator-ui".

Reply:
Fixed!